packages/transport/src/utils/receive.ts

```diff
@@ -3,29 +3,45 @@ import { Root } from 'protobufjs/light';
 import { decode as decodeProtobuf, createMessageFromType } from '@trezor/protobuf';
 import { TransportProtocol } from '@trezor/protocol';
 
-async function receiveRest(
+import { success, error } from './result';
+import { AbstractApi } from '../api/abstract';
+import * as ERRORS from '../errors';
+
+async function receiveRest<T extends () => ReturnType<AbstractApi['read']>>(
     result: Buffer,
-    receiver: () => Promise<Buffer>,
+    receiver: T,
     offset: number,
     expectedLength: number,
     chunkHeader: Buffer,
-): Promise<void> {
+) {
     if (offset >= expectedLength) {
-        return;
+        return error({ error: ERRORS.UNEXPECTED_ERROR, message: 'offset >= expected length' });
```

Comment:
this is somewhere in between unexpected and expected error. it really shouldn't not happen, its just a sanity check. so I am not sure whether it should be strongly typed or not.

Reply:
or maybe let it throw